src/cargo/core/workspace.rs

```diff
@@ -1669,6 +1671,7 @@ impl InheritableFields {
         publish: Option<VecStringOrBool>,
         edition: Option<String>,
         badges: Option<BTreeMap<String, BTreeMap<String, String>>>,
+        ws_path: PathBuf,
```

Comment:
In cargo-add, I instead make everything absolute "at rest" and then just do the diffing later
- Benefits:
  - One less thing to pass around and track
  - When caching, we can share the `join` across all workspace members but this is probably a drop in the bucket of work we do
- Downsides:
  - We make things absolute when they might not be used (which if they aren't used, then the user likely won't have them in the file)
  - if the existing types can't handle it well (especially if the path has to be put in a `str` then the whole thing has to be UTF-8 and not just the tail end ... but other parts of cargo might be enforcing this)

Reply:
I'm not quite sure what you mean by this

Reply:
Instead of adding a `ws_path` member to the struct, have `new` do
`license_file: license_file.map(|l| ws_path.join(l))`

(of course, there are more details to the implementation)

Reply:
How would this work for dependencies? I don't think there is a great way to deal with those as we parse them. I could be missing something but they are parsed into a `DetailedTomlDependency` without a way for us to specify the workspace path.

Reply:
`new` could iterate over the dependencies, adding the workspace path to them.

Reply:
I tried doing this but didn't love the outcome. If you still feel its better that's fine with me.

Reply:
This was brainstorming.  We can drop it.